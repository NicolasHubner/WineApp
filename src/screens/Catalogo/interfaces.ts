export interface ITems {
    id: number;
    name: string;
    price: number;
    priceMember: number;
    priceNonMember: number;
    discount: number;
    image: string;
  }
 export interface IData {
    totalItems: number;
    items: ITems[];
  }