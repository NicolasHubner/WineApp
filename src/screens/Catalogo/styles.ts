import styled from 'styled-components/native';


export const CardView = styled.TouchableOpacity`
  width: 100%;
  flex-direction: column;
  justify-content: space-evenly;
  align-items: center;
  background-color: #FFFFFF;
  height: 348px;
  margin-bottom: 8px;
  border-radius: 4px;
  padding-top: 15px;
  box-shadow: 0px 1px 2px rgba(0, 0, 0, 0.2);

`
export const WineImage = styled.Image`
  width: 100px;
  height: 200px;
`
export const StickImage = styled.Image`
  position: relative;
  bottom: 175px;
  right: 60px;

`
export const MemberPrice = styled.Text`
  font-family: 'Lato';
  width: 35px;
  /* height: 2px; */
  font-size: 10px;
  font-weight: 300;
  text-align: center;
`
export const TtitleText = styled.Text`
  width: 90%;
  font-family: 'Lato';
  text-align: center;
  margin-bottom: 12px;
  font-weight: 700;
  line-height: 17px;
`
export const ViewPriceBoth = styled.View`
  flex-direction: row;
  width: 80%;
  justify-content: space-evenly;
  /* margin-top: 5px; */
  margin-bottom: 5px;
  align-items: center;
`
export const TextNoMember = styled.Text`
  font-size: 11px;
  font-family: 'Lato';
  font-weight: 400;
  color: #888888;
`
export const PriceMemberPink = styled.Text`
  font-size: 18px;
  font-family: 'OpenSans';
  color: #C81A78;
  font-weight: 700;
`
export const PriceStrached = styled.Text`
  font-family: 'Lato';
  font-weight: 400;
  font-size: 12px;
  line-height: 12px;
  align-items: flex-end;
  color: #888888;
  text-decoration: line-through;
`
export const Discount = styled.Text`
  background-color: #F79552;
  border-radius: 2px;
  width: 65px;
  height: 22px;
  color: white;
  text-align: center;
  padding: 1px 4px 1px 4px;
  /* align-items: center;
  justify-content: center; */
`
export const CounterItems = styled.Text`
  margin-left: 30px;
  font-family: 'NeoSans';
  font-weight: 400;
  font-size: 16px;
  line-height: 24px;
  color: #555555;
`
export const InputSection = styled.View`
flex-direction: row;
justify-content: flex-start;
align-items: center;
/* border-width: 0.5;
border-color: black; */
height: 40px;
border-radius: 5px;
padding: 10px;
background-color: #EAEAEA;
margin: 0 auto;
margin-top: 32px;
width: 90%;
margin-bottom: 24px;
`

export const CardContainer = styled.View`
  height: 450px;
  width: 180px;
  flex-direction: column;
  align-items: flex-start;
  justify-content: center;
  /* text-align: center; */
  /* flex: 1; */
`
