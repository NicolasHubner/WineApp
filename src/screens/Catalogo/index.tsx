import React, { useCallback, useContext, useEffect, useState } from 'react';
import { SafeAreaView, Text, Image, TextInput } from 'react-native';
import { Button, Input } from 'react-native-elements';
import { ScrollView } from 'react-native-gesture-handler';
import { IData, INavigation } from './interfaces';
import { CardView, WineImage, StickImage, MemberPrice, TtitleText, ViewPriceBoth, TextNoMember, PriceMemberPink, PriceStrached, Discount, CounterItems, InputSection, CardContainer } from './styles';
import Selos from '../../assets/selos.png';
import { useFonts } from 'expo-font';
import SearchBar from '../../assets/Search.png'
import { useNavigation, NavigationContainer } from '@react-navigation/native';
import { UserContext, IValue } from '../../context/UserContext';
import { PriceConvert } from '../../helpers/priceConverter';



export default function Catalogo() {
  const [data, setData] = useState<IData | null>(null);
  const navigation = useNavigation() as INavigation;
  const { dispatch } = useContext(UserContext) as unknown as IValue;

  const [fontsLoaded] = useFonts({
    'Lato': require('../../assets/fonts/Lato-Regular.ttf'),
    'OpenSans': require('../../assets/fonts/open-sans-v34-latin-regular.ttf'),
    'NeoSans': require('../../assets/fonts/NeoSansPro-Regular.ttf')
  });


  useEffect(() => {
    const dataWines = async () => {
      const res = await fetch('https://wine-back-test.herokuapp.com/products?page=1&limit=45')
      const data = await res.json();
      setData(data);
    }
    dataWines()
  }, [])
  return (
      <SafeAreaView style={{ marginBottom: 100 }}>
        <InputSection>
          <Image
            source={SearchBar} />
          <TextInput style={{ marginLeft: 18 }} placeholder='O que você está procurando?' />
        </InputSection>
        <CounterItems>{data?.totalItems} produtos encontrados</CounterItems>
        <ScrollView contentContainerStyle={{ flexDirection: 'row', flexWrap: 'wrap', justifyContent: 'space-evenly' }}>
          {data?.items.map((item) => (
            <CardContainer key={item.id}>
              <CardView
                onPress={() => navigation.navigate({
                  name: 'PageProduct',
                  params: {
                    item: item
                  }
                })}
              >
                <WineImage
                  source={{ uri: item.image }}
                />
                <TtitleText>{item.name}</TtitleText>
                <ViewPriceBoth>
                  <PriceStrached>R$ {PriceConvert(item.priceNonMember)}</PriceStrached>
                  <Discount>{Number((item.discount / item.priceNonMember).toFixed(2)) * 100}% OFF</Discount>
                </ViewPriceBoth>
                <ViewPriceBoth style={{ marginBottom: 20, marginTop: 10 }}>
                  <MemberPrice>SÓCIO WINE</MemberPrice>
                  <PriceMemberPink>R$ {PriceConvert(item.priceMember)}</PriceMemberPink>
                </ViewPriceBoth>
                <TextNoMember style={{ marginBottom: -12 }}>NÃO SÓCIO R$ {PriceConvert(item.priceNonMember)}</TextNoMember>
                <StickImage
                  source={Selos}
                />
              </CardView>
              <Button
                buttonStyle={{
                  width: "100%",
                  backgroundColor: '#7EBC43',
                  justifyContent: "flex-end",
                  paddingRight: 58,
                  borderRadius: 4,
                  height: 40
                }}
                title="Adicionar"
                titleStyle={{ fontFamily: "Lato", fontSize: 14, textAlign: 'center', fontWeight: "700", lineHeight: 16 }}
                onPress={() => dispatch({
                  type: 'SetCounter'
                })}
              />

            </CardContainer>
          ))}
        </ScrollView>
      </SafeAreaView>
  )
}