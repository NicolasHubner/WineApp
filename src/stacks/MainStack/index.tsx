import React from "react";
import { createStackNavigator } from '@react-navigation/stack';
import { NavigationContainer } from '@react-navigation/native';
import Catalogo from "../../screens/Catalogo";
import PageProduct from "../../screens/PageProduct";
import CustomStack from '../../components/HeaderCustom/CustomLeft/index';
import CustomRight from "../../components/HeaderCustom/CustomRight";
import { navigationRef, isReadyRef } from "../../RootNavigation";

const Stack = createStackNavigator();

export default function MyStack() {
  return (
    <NavigationContainer ref={navigationRef}>{
      <Stack.Navigator
        initialRouteName="Home"
        screenOptions={{
          cardStyle: { backgroundColor: "#DEDEDE" },
          headerStyle: { backgroundColor: "#FFFFFF" },
          headerRight: () => <CustomRight />
        }}

      >
        <Stack.Screen
          options={{ title: "", headerLeft: () => <CustomStack />, }}
          name="Home" component={Catalogo} />
        <Stack.Screen
          name="PageProduct"
          options={{
            title: "", headerStyle: { backgroundColor: '#DEDEDE' }
          }}
          component={PageProduct as any} />
      </Stack.Navigator>
}</NavigationContainer>
  )
}