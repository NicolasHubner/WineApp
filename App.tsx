import { StatusBar } from 'expo-status-bar';
import 'react-native-gesture-handler';
// import { NavigationContainer } from '@react-navigation/native';
import MainStack from './src/stacks/MainStack'
import UserContextProvider from './src/context/UserContext'
import { Platform } from 'react-native';


if (Platform.OS === 'android') {
  require('intl');
  require('intl/locale-data/jsonp/pt-BR');
}

export default function App() {
  return (
    <UserContextProvider>
      {/* <NavigationContainer>{ */}
        <MainStack />
        <StatusBar />
      {/* } */}
      {/* </NavigationContainer> */}
    </UserContextProvider>
  )
}