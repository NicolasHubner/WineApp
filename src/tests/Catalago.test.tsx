import React from 'react'
import renderer, { act } from 'react-test-renderer';
import { render, screen } from '@testing-library/react-native';
// import Catalogo from '../screens/Catalogo';
import MainStack from '../stacks/MainStack'
import App from '../../App'
import Catalogo from '../screens/Catalogo';

describe('App', () => {
  it('Renderizar o componente raiz', () => {
    const tree = renderer.create(<App/>).toJSON();
    expect(tree).toMatchSnapshot();
  })
})

describe('Catalogo', () => {
  it('Texto do input', () => {
    // act(() => render(<Catalogo/>))
    render(<App/>)
    const inputText = screen.getByPlaceholderText('O que você está procurando?')
    // console.log(inputText);
    expect(inputText).toMatch
  })
  it('Texto de produtos encontrados', () => {
    render(<App/>)
    const textFindProduct = screen.getByText('produtos encontrados')
    expect(textFindProduct).toMatch
  })
  // it('Texto sobre quantidade certas de produtos', () => {
  //   render(<App/>)
  //   const numberFindProduct = screen.getByText('45');
  //   expect(numberFindProduct).toMatch
  // })
})

describe('MainStack', () => {
  it('Render MainStack', () => {
    render(<MainStack/>)
  })
  it('Aparecer os 45 produtos encontrados do fetch', async() => {
    render(<MainStack/>)
    const textProducts = await screen.findByText('45 produtos encontrados')
    expect(textProducts).toMatch;
  })
})