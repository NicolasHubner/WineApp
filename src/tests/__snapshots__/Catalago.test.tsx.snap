// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Renderizar o componente raiz 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <RNCSafeAreaProvider
    onInsetsChange={[Function]}
    style={
      Array [
        Object {
          "flex": 1,
        },
        undefined,
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "backgroundColor": "rgb(242, 242, 242)",
            "flex": 1,
          },
          undefined,
        ]
      }
    >
      <View
        collapsable={false}
        pointerEvents="box-none"
        style={
          Object {
            "zIndex": 1,
          }
        }
      >
        <View
          accessibilityElementsHidden={false}
          importantForAccessibility="auto"
          onLayout={[Function]}
          pointerEvents="box-none"
          style={null}
        >
          <View
            collapsable={false}
            pointerEvents="box-none"
            style={
              Object {
                "bottom": 0,
                "left": 0,
                "opacity": 1,
                "position": "absolute",
                "right": 0,
                "top": 0,
                "zIndex": 0,
              }
            }
          >
            <View
              collapsable={false}
              style={
                Object {
                  "backgroundColor": "#FFFFFF",
                  "borderBottomColor": "rgb(216, 216, 216)",
                  "flex": 1,
                  "shadowColor": "rgb(216, 216, 216)",
                  "shadowOffset": Object {
                    "height": 0.5,
                    "width": 0,
                  },
                  "shadowOpacity": 0.85,
                  "shadowRadius": 0,
                }
              }
            />
          </View>
          <View
            collapsable={false}
            pointerEvents="box-none"
            style={
              Object {
                "height": 44,
                "maxHeight": undefined,
                "minHeight": undefined,
                "opacity": undefined,
                "transform": undefined,
              }
            }
          >
            <View
              pointerEvents="none"
              style={
                Object {
                  "height": 0,
                }
              }
            />
            <View
              pointerEvents="box-none"
              style={
                Object {
                  "alignItems": "stretch",
                  "flex": 1,
                  "flexDirection": "row",
                }
              }
            >
              <View
                collapsable={false}
                pointerEvents="box-none"
                style={
                  Object {
                    "alignItems": "flex-start",
                    "flexBasis": 0,
                    "flexGrow": 1,
                    "justifyContent": "center",
                    "marginStart": 0,
                    "opacity": 1,
                  }
                }
              >
                <View>
                  <Image
                    source={1}
                    style={
                      Object {
                        "marginLeft": 16,
                      }
                    }
                  />
                </View>
              </View>
              <View
                collapsable={false}
                pointerEvents="box-none"
                style={
                  Object {
                    "justifyContent": "center",
                    "marginHorizontal": 16,
                    "maxWidth": 590,
                    "opacity": 1,
                  }
                }
              >
                <Text
                  accessibilityRole="header"
                  aria-level="1"
                  collapsable={false}
                  numberOfLines={1}
                  onLayout={[Function]}
                  style={
                    Object {
                      "color": "rgb(28, 28, 30)",
                      "fontSize": 17,
                      "fontWeight": "600",
                    }
                  }
                />
              </View>
              <View
                collapsable={false}
                pointerEvents="box-none"
                style={
                  Object {
                    "alignItems": "flex-end",
                    "flexBasis": 0,
                    "flexGrow": 1,
                    "justifyContent": "center",
                    "marginEnd": 0,
                    "opacity": 1,
                  }
                }
              >
                <View>
                  <Image
                    source={1}
                    style={
                      Object {
                        "marginRight": 16,
                        "marginTop": 48,
                      }
                    }
                  />
                  <Text
                    style={
                      Object {
                        "backgroundColor": "white",
                        "color": "#00B495",
                        "marginLeft": 42,
                        "marginTop": 85,
                        "position": "absolute",
                      }
                    }
                  >
                    0
                  </Text>
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
      <RNSScreenContainer
        onLayout={[Function]}
        style={
          Object {
            "flex": 1,
          }
        }
      >
        <RNSScreen
          activityState={2}
          collapsable={false}
          forwardedRef={[Function]}
          gestureResponseDistance={
            Object {
              "bottom": -1,
              "end": -1,
              "start": -1,
              "top": -1,
            }
          }
          pointerEvents="box-none"
          style={
            Object {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            }
          }
        >
          <View
            collapsable={false}
            style={
              Object {
                "opacity": 1,
              }
            }
          />
          <View
            accessibilityElementsHidden={false}
            closing={false}
            gestureVelocityImpact={0.3}
            importantForAccessibility="auto"
            onClose={[Function]}
            onGestureBegin={[Function]}
            onGestureCanceled={[Function]}
            onGestureEnd={[Function]}
            onOpen={[Function]}
            onTransition={[Function]}
            pointerEvents="box-none"
            style={
              Array [
                Object {
                  "display": "flex",
                  "overflow": undefined,
                },
                Object {
                  "bottom": 0,
                  "left": 0,
                  "position": "absolute",
                  "right": 0,
                  "top": 0,
                },
              ]
            }
            transitionSpec={
              Object {
                "close": Object {
                  "animation": "spring",
                  "config": Object {
                    "damping": 500,
                    "mass": 3,
                    "overshootClamping": true,
                    "restDisplacementThreshold": 10,
                    "restSpeedThreshold": 10,
                    "stiffness": 1000,
                  },
                },
                "open": Object {
                  "animation": "spring",
                  "config": Object {
                    "damping": 500,
                    "mass": 3,
                    "overshootClamping": true,
                    "restDisplacementThreshold": 10,
                    "restSpeedThreshold": 10,
                    "stiffness": 1000,
                  },
                },
              }
            }
          >
            <View
              collapsable={false}
              pointerEvents="box-none"
              style={
                Object {
                  "flex": 1,
                }
              }
            >
              <View
                collapsable={false}
                forwardedRef={[Function]}
                handlerTag={1}
                handlerType="PanGestureHandler"
                needsOffscreenAlphaCompositing={false}
                onGestureHandlerEvent={[Function]}
                onGestureHandlerStateChange={[Function]}
                style={
                  Object {
                    "flex": 1,
                    "transform": Array [
                      Object {
                        "translateX": 0,
                      },
                      Object {
                        "translateX": 0,
                      },
                    ],
                  }
                }
              >
                <View
                  style={
                    Array [
                      Object {
                        "flex": 1,
                        "overflow": "hidden",
                      },
                      Array [
                        Object {
                          "backgroundColor": "rgb(242, 242, 242)",
                        },
                        Object {
                          "backgroundColor": "#DEDEDE",
                        },
                      ],
                    ]
                  }
                >
                  <View
                    style={
                      Object {
                        "flex": 1,
                        "flexDirection": "column-reverse",
                      }
                    }
                  >
                    <View
                      style={
                        Object {
                          "flex": 1,
                        }
                      }
                    >
                      <RCTSafeAreaView
                        emulateUnlessSupported={true}
                        style={
                          Object {
                            "marginBottom": 100,
                          }
                        }
                      >
                        <View
                          style={
                            Array [
                              Object {
                                "alignItems": "center",
                                "backgroundColor": "#EAEAEA",
                                "borderRadius": 5,
                                "flexDirection": "row",
                                "height": 40,
                                "justifyContent": "flex-start",
                                "marginBottom": 24,
                                "marginLeft": "auto",
                                "marginRight": "auto",
                                "marginTop": 32,
                                "paddingBottom": 10,
                                "paddingLeft": 10,
                                "paddingRight": 10,
                                "paddingTop": 10,
                                "width": "90%",
                              },
                            ]
                          }
                        >
                          <Image
                            source={1}
                          />
                          <TextInput
                            onChangeText={[Function]}
                            placeholder="O que você está procurando?"
                            style={
                              Object {
                                "marginLeft": 18,
                              }
                            }
                          />
                        </View>
                        <Text
                          style={
                            Array [
                              Object {
                                "color": "#555555",
                                "fontFamily": "sans-serif",
                                "fontSize": 16,
                                "fontWeight": "400",
                                "lineHeight": 24,
                                "marginLeft": 30,
                              },
                            ]
                          }
                        >
                           produtos encontrados
                        </Text>
                        <RCTScrollView
                          collapsable={false}
                          contentContainerStyle={
                            Object {
                              "flexDirection": "row",
                              "flexWrap": "wrap",
                              "justifyContent": "space-evenly",
                            }
                          }
                          handlerTag={2}
                          handlerType="NativeViewGestureHandler"
                          onGestureHandlerEvent={[Function]}
                          onGestureHandlerStateChange={[Function]}
                        >
                          <View />
                        </RCTScrollView>
                      </RCTSafeAreaView>
                    </View>
                  </View>
                </View>
              </View>
            </View>
          </View>
        </RNSScreen>
      </RNSScreenContainer>
    </View>
  </RNCSafeAreaProvider>
</View>
`;
